client/src/components/History/Content/GenericElement.vue

```diff
@@ -1,36 +1,47 @@
+<script setup lang="ts">
+import Vue from "vue";
+import ContentItem from "./ContentItem.vue";
+import { ref } from "vue";
+
+interface DatasetCollectionElement {
+    id: string;
+    element_index: number;
+    element_identifier: string;
+    element_type: string;
+    object: Record<string, any>;
```

Comment:
Avoid `any` types.

This seems to be a recursive type:

```suggestion
    object: DatasetCollection ;
```

Reply:
It is a recursive type and we have an (almost) correct type in schema/components/DCObject, maybe that should be used ?

Reply:
If it's an API type, that might be a good idea.
In general I think duck typing is also fine though, if the number of fields the component requires is just a few. Since this object is quite clearly an API response and the overlap seems to be fairly large, I'd lean towards using the existing type here.

Reply:
Agreed on that too, but I think the dispatched child components may need to know the child types (eventually, once they have type hints on their props).

Reply:
Might have to use `schema/components/DCObject`, since although it is recursive, it is not `DatasetCollection`, the object can also be a `Dataset`.

Reply:
You mean for getting types out of the schema ? https://github.com/mvdbeek/galaxy/blob/746e219f7b53186d0b3f75e0f69d397bd13ebb43/client/src/components/History/Export/services.ts#L7 or https://github.com/mvdbeek/galaxy/blob/746e219f7b53186d0b3f75e0f69d397bd13ebb43/client/src/components/Datatypes/model.ts#L3

Reply:
Thx, wanted to make sure it's the same way.